SaveEarth/Projects/SaveEarth/SaveEarth/Sources/UserInterface/Screens/MainView/MainView.swift

```diff
@@ -9,101 +9,132 @@ import ComposableArchitecture
+      let clearedCount = missionList.filter(\.isClear).count
+      return Float(clearedCount) / Float(missionList.count)
+    } else {
+      return .zero
+    }
+  }
 
-  var content: some View { // UI 그리기
+  var body: some View {
     ZStack {
       MapView(
-        latitude: store.state.latitude,
-        longitude: store.state.longitude
+        latitude: $latitude,
+        longitude: $longitude
       )
       .overlay {
-        TemperatureGradient(complete: store.state.completeRate)
-          .ignoresSafeArea()
-      }
-      .overlay {
-        if let missions = store.dayInfo?.missionList, missions.allSatisfy(\.isClear) {
-          CompleteQuestView()
-        } else {
-          Image(systemName: "flame.fill")
-            .foregroundColor(.red)
-            .font(.largeTitle)
-        }
-      }
-      if let dayInfo = store.dayInfo {
-        VStack {
-          Text(store.state.completeRate != 1 ? "뜨거운 지구를 구해주세요!! 😱" : "오늘도 지구를 조금 살려냈어요!")
-            .font(.title)
-            .padding(.vertical)
-          Text("Temperature 🌡️")
-            .font(.title2)
-            .padding()
-          HStack {
-            Spacer()
-            VStack(alignment: .leading) {
-              Text("현재온도")
-              Text(String(format: "%.2f℃", dayInfo.temperatureData.currentTemperature))
+        TemperatureGradient(complete: completeRate)
+          .overlay(alignment: .center) {
+            if let isCleared = isCleared {
+              if isCleared {
+                CompleteQuestView()
+              } else {
+                Image(systemName: "flame.fill")
+                  .foregroundColor(.red)
+                  .font(.largeTitle)
+              }
             }
-            Spacer()
-            VStack(alignment: .leading) {
-              Text("평균온도")
-              Text(String(format: "%.2f℃", dayInfo.temperatureData.historicTemperature))
-            }
-            Spacer()
-          }
-          .font(.title2)
-          Spacer()
-          if !dayInfo.missionList.allSatisfy(\.isClear) {
-            QuestFloatingButton(
-              numberOfQuests: UInt(dayInfo.missionList.filter { !$0.isClear }.count)
-            ) {
-              store.send(.questFloatingButtonTapped)
-            }
-            .transition(AppearingTransition())
-            .animation(.spring(), value: store.state.destination)
           }
-        }
+          .ignoresSafeArea()
+      }
+      if let dayInfo = dayInfo {
+        MainInterface(
+          dayInfo: dayInfo,
+          completeRate: completeRate
+        )
       } else {
         ProgressView()
       }
     }
-    .sheet(
-      item: $store.scope(
-        state: \.destination?.modal,
-        action: \.destination.modal
-      )
-    ) { store in
-      MissionListModal(store: store)
-        .presentationDetents([.height(260)])
+    .task {
+      do {
+        try await updateCurrentLocationTemperature()
+      } catch {
+
+      }
```

Comment:
catch 구문이 필욯하지 않다면, 아래와 같이 코드를 작성해도 좋을 것 같습니다!
```suggestion
      try? await updateCurrentLocationTemperature()
```

Reply:
에러처리 설계 전이라서 비웠는데

현시점에서는 말씀해주신게 좋은것 같아요!